src/types.ts

```diff
@@ -3,6 +3,22 @@ import { Consumer } from "sqs-consumer";
 import { MessageAttributeValue, SQSClientConfig, Message } from "@aws-sdk/client-sqs";
 import { SNSClientConfig } from "@aws-sdk/client-sns";
 import { LambdaClientConfig } from "@aws-sdk/client-lambda";
+import { AsyncLocalStorage } from "async_hooks";
+
+export interface Logger {
+  error(error: any): void;
+  warn(message: any): void;
+  debug(message: any): void;
+  info(message: any): void;
+  /**
+   * @returns Returns the async local storage and store which will be used to call the message handler
+   * so that the other log functions and those logs that the service itself does happen in the same context
+   */
+  getStore?: (executionContext?: ProcessMessageContext) => ({
```

Comment:
I think rather than leaving the client to provide store we can create and run inside an async context from within if the log level is set to trace
Thought?

Reply:
Yes but the only problem is that whatever logs the client code makes will be not be in the same context and hence lack the trace execution id

Reply:
Since we are given nothing additional but only the original payload in the listener functions

Reply:
What you state we could add it for the default logger

Reply:
I guess let's have an optional 2nd argument in the listener function with the following interface:

```
interface MessageMetaData {
executionTraceId: string;
messageId: string;
messageAttributes: IMessageAttributes;
...otherMetaProps
}
```

Then the consumers can use executionTraceId from here if they would like to